test/blockdiagonal.jl

```diff
@@ -90,4 +90,13 @@ using Test
         @test Matrix(b1) ≈ b1
         @test b1 ≈ Matrix(b1)
     end  # Equality
+
+    @testset "Non-Square Matrix" begin
```

Comment:
We should probably be testing non-square block diagonals in the matrix multiplication tests too (in tezt/base_maths.jl)